Tests/Unit/PollerTests.swift

```diff
@@ -12,44 +12,51 @@ class PollerSpec: QuickSpec {
                     self.addedTimer = timer
                 }
             }
-            it("creates a timer with time interval equal to default delay of 1 hour") {
+            it("creates a timer with time interval equal to default delay when no explicit delay param is passed into initializer") {
                 let runLoop: MockRunLoop = MockRunLoop()
                 let poller = Poller(runLoop: runLoop)
 
                 poller.start { }
 
-                expect(runLoop.addedTimer?.timeInterval).toEventually(equal(3600.0))
+                expect(runLoop.addedTimer?.timeInterval).toEventually(equal(PollerConstants.defaultDelay))
             }
 
-            it("creates a timer with time interval equal to delay param") {
+            it("creates a timer with time interval equal to passed in delay param when param value is greater than minimum delay") {
                 let runLoop: MockRunLoop = MockRunLoop()
-                let poller = Poller(delay: 10.5, runLoop: runLoop)
+                let poller = Poller(delay: 100.5, runLoop: runLoop)
 
                 poller.start { }
 
-                expect(runLoop.addedTimer?.timeInterval).toEventually(equal(10.5))
+                expect(runLoop.addedTimer?.timeInterval).toEventually(equal(100.5))
             }
 
-            it("synchronously executes action closure exactly once") {
-                var actionCalled = 0
-                let poller = Poller()
+            it("creates a timer with time interval equal to minimum delay if passed in delay param is zero") {
+                let runLoop: MockRunLoop = MockRunLoop()
+                let poller = Poller(delay: 0, runLoop: runLoop)
 
-                poller.start {
-                    actionCalled += 1
-                }
+                poller.start { }
 
-                expect(actionCalled).to(equal(1))
+                expect(runLoop.addedTimer?.timeInterval).toEventually(equal(PollerConstants.minimumDelay))
             }
 
-            it("asynchronously executes the action after the specified delay") {
+            it("creates a timer with time interval equal to minimum delay if passed in delay param is less than minimum delay") {
+                let runLoop: MockRunLoop = MockRunLoop()
+                let poller = Poller(delay: 10, runLoop: runLoop)
+
+                poller.start { }
+
+                expect(runLoop.addedTimer?.timeInterval).toEventually(equal(PollerConstants.minimumDelay))
+            }
+
+            it("synchronously executes action closure exactly once") {
```

Comment:
synchronously because RunLoop.current is used? If so then maybe it's better to add this info in `context()` or just in test name

Reply:
When poller start is called passing in an action closure, it calls the closure once then sets up a timer to call the action closure after a delay. So if there's a synchronous call to `start` the action should happen once and once only.

Reply:
Ah ok, I get it now. 👍